### ISSUE TITLE
Cannot use Dates as query params

### ISSUE THREAD

**Issue by omri-benarye-cyera** [box/box-typescript-sdk-gen#223]
- [x] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [x] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [x] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [x] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue
Trying to use `getEvents` functionality:
```
  const events = await client.events.getEvents({
    streamType: 'admin_logs' as GetEventsQueryParamsStreamTypeField,
    limit: 10,
    eventType: ['UPLOAD'],
    createdAfter: { value: new Date('2024-06-09T10:53:43-08:00') },
  });
```

I am getting:
```
{
    "type": "error",
    "status": 400,
    "code": "bad_request",
    "help_url": "http://developers.box.com/docs/#errors",
    "message": "Invalid created_after parameter",
    "request_id": "du1a5ehqkfz86qkm"
}
```

Tried to do the same with CURL I am getting the same for `created_after` with `+` in the timestamp, BUT - it does work with a `-`.
example:
created_after: 2024-06-10T01:31:53-01:00 - WORK
created_after: 2024-06-10T01:31:53+01:00 - DOES NOT WORK

In this SDK I saw you have this function:
dateTimeToString(dateTime: DateTime): string {
  return dateTime.value.toISOString().match(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}/)![0] + '+00:00';
}
That adds this `+`, which always failing the usage of it.
 
### Versions Used

Typescript SDK: 1.0.0
Platform: typescript v4.8.4

[sdk-docs]: https://github.com/box/box-typescript-sdk-gen/tree/main/docs
[api-docs]: https://developer.box.com/docs
[dev-forums]: https://community.box.com/t5/Platform-and-Development-Forum/bd-p/DeveloperForum
[github-repo]: https://github.com/box/box-typescript-sdk-gen/search?type=Issues

**Comment by mwwoda** [comment]
Hi @omri-benarye-cyera 
As you pointed out, it looks like this endpoint have some problem with positive offsets. Let me reach out to service owners to confirm it

**Comment by mwwoda** [comment]
After further investigation it looks like this could be matter of encoding. `+` sign is interpreted as white space when used in the URL. You should encode it to the form of `%2B`. Then the cURL request works. I guess the SDK should do the same.

**Comment by omri-benarye-cyera** [comment]
Tried using `%2B` with CURL and it does work!
So I guess dateTimeToString should be edited - to have `'%2B00:00'` instead of `'+00:00'`.
(Tried to open PR for it but I do not have permissions)

**Comment by mwwoda** [comment]
Actually the root cause is a bit different in the SDK. In case datetime was used as query param, it included quotes that caused a problem with the request.

This SDK is actually generated based on our custom codegen engine which is currently private. I just submitted a fix for it, so it should be downstreamed to this repo soon.

**Comment by mwwoda** [comment]
Fixed by #255. Fix should be released with the new version of the SDK.

**Comment by omri-benarye-cyera** [comment]
Thank you @mwwoda for the fast resolve!
